### PR TITLE
Export Config with `--generate-config`

### LINKED ISSUE #48: Generate default yaml configuration on cli flag

### PR DESCRIPTION
**Pull request by lummax**
Fixes #48.

`--generate-config` is marked as `help=True` to prevent the required option exception. But the `jcommander.Parameter` class only cares for the existence of at least one of those and there is no further special treatment.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -157,7 +157,7 @@ allprojects {
 }
 
 detekt {
-	config = "$project.projectDir/default-detekt-config.yml"
+	config = "$project.projectDir/detekt-cli/src/main/resources/default-detekt-config.yml"
```
Comment (arturbosch) [inline build.gradle]: Thats wrong. The config parameter should be used to locate your default-config from any path on your filesystem. If you have the default-config inside your resources (maven multi-module support), you should use the `configResource` field eg. `configResource = "default-detekt-config.yml"` which will be extracted from the resources

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Configurations.kt [hunk 2]
```diff
@@ -37,4 +38,16 @@ class FormatConfig(private val useTabs: Boolean) : Config {
...
 	}
+}
+
+/**
+ * @author lummax
+ */
+class ConfigExporter(val main: Main) {
+
+	fun execute() {
+		val defaultConfig = ClassLoader.getSystemClassLoader().getResourceAsStream("default-detekt-config.yml")
```
Comment (arturbosch) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Configurations.kt]: Use `javaClass.getResourceAsStream()` and explicitly throw an InvalidStateException if the inputstream is null

File: detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt [hunk 3]
```diff
@@ -64,6 +68,9 @@ class Main {
 			if (main.debug) {
 				Debugger(main).execute()
 				return
+			} else if (main.generateConfig) {
```
Comment (arturbosch) [inline detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Main.kt]: I think it's time to introduce an `Executable` interface :) Something like:

```
val executable = when {
    main.debug -> Debugger(main) 
    main.generateConfig -> ConfigExporter(main)
    else -> Runner(main)
}
executable.execute()
```

**Comment by arturbosch** [comment]
The review fixes and I'm gladly merging this PR. Thx!

File: build.gradle (hunk 1 above)
Comment (lummax) [inline build.gradle]: This is just the local `build.gradle`. From what I undernstand this is the local configuration of the gradle detekt plugin? Within this context, this path will always contain the configuration. Using `configResource` does not work here, as the `default-detekt-config.yml` only gets bundled into the `detekt-cli` jar?

Comment (arturbosch) [inline build.gradle]: Ah, your right. I somehow thought this was the README and the users would get confused. :)

**Pull request merged**